### DISCUSSION TITLE
How does work postcss with tailwind postcss plugin ?

### DISCUSSION THREAD

**Discussion by RomainoSoko** [tailwindlabs/tailwindcss#5061] [Help]
Hi there ! 

I recently create a new project using webpack 5. I've setup tailwind and everything work's good. 

I've also setup storybook on my project, and got some issue because of the novelty of it webpack 5 compatibility (some addons override webpack version). So i decided to add to the .storybook config folder its dedicated package.json and node_modules for preventing webpack version's override. After that, I've been surprised to see that the tailwind config that is at the root of my projects was'nt compiled with storybook (so i hadn't my utility classes).  

So, I've added a postcss.config.js (using tailwind and auto-prefixer) file inside the .storybook config folder and related it to my storybook's webpack config like this :

```
file : app/.storybook/config/main.js

use: [
        {
          loader: 'postcss-loader',
          options: {
            postcssOptions: {
              config: path.resolve(__dirname, "./../postcss.config.js"),
            },
          }
        }
      ],
```

```
file: app/.storybook/postcss.config.js

module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};

```

Thought, postcss will take the tailwind.config.js file at the root of my project (app/tailwind.config.js). But It was still not working until i add its own tailwind.config.js to the .storybook folder. 

Its like it need a tailwind.config.js at the same root that postcss.config.js file.

That made me thought that I do not understand how postcss is working with tailwind.

So my question is : Is that possible to know how the tailwind-postcss plugin is working and is that possible to specify the tailwind config we want to use (because i want to use the one that is at the root of my project, not inside the .storybook config folder) ?

Thanks !

**Comment by Jerry004** [comment]
One of the most useful features preprocessors offer is the ability to organize your CSS into multiple files and combine them at build time by processing @import statements in advance, instead of in the browser.

[Myloweslife](https://www.myloweslife.win/)
